x-pack/plugins/file_data_visualizer/public/application/components/stats_table/components/field_data_expanded_row/keyword_content.tsx

```diff
@@ -5,21 +5,55 @@
  * 2.0.
  */
 
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import type { FieldDataRowProps } from '../../types/field_data_row';
 import { TopValues } from '../../../top_values';
+import { EMSTermJoinConfig } from '../../../../../../../maps/public';
+import { useFileDataVisualizerKibana } from '../../../../kibana_context';
 import { DocumentStatsTable } from './document_stats';
 import { ExpandedRowContent } from './expanded_row_content';
+import { ChoroplethMap } from './choropleth_map';
+
+const COMMON_EMS_LAYER_IDS = [
+  'world_countries',
+  'administrative_regions_lvl2',
+  'usa_zip_codes',
+  'usa_states',
+];
 
 export const KeywordContent: FC<FieldDataRowProps> = ({ config }) => {
-  const { stats } = config;
+  const [EMSSuggestion, setEMSSuggestion] = useState<EMSTermJoinConfig | null | undefined>();
+  const { stats, fieldName } = config;
   const fieldFormat = 'fieldFormat' in config ? config.fieldFormat : undefined;
+  const {
+    services: { maps: mapsPlugin },
+  } = useFileDataVisualizerKibana();
+
+  const getEMSTermSuggestions = async () => {
```

Comment:
this function doesn't return anything, and so calling it `get....` might be a bit misleading.
perhaps `loadEMSTermSuggestions` would be better?

Could this be wrapped in a `useCallback`?